execution_workflows/APA-Scan/workflow/Snakefile

```diff
@@ -0,0 +1,337 @@
+# dirs/files of interest, which are set at the top. Output is just set to the
+# curren working directory as APA-Scan again is really painful and picky so
+# best not to tempt fate and the bioinformatics demons.
+
+# It's s obviously possible to just
+# define the configuration.ini file before running the whole pipeline, but this
+# wont work for multiple runs with different samples and studies. This is my
+# attempt to make it a bit generalisable (touch wood). I'm hoping whatever levels
+# you want to compare, you could parse that in some way from the previous step?
+
+################################################################################
+
+
+
+rule generate_configurationfile:
+    """Generate configuration.ini file for input to run_apa_scan
+    """
+    input:
+        "blank_configuration.ini"
+    output:
+        "configuration.ini"
+    threads: THREADS
+    conda: os.path.join(config["envs"], CONDA_YAML)
+    group: "APA_SCAN"
+    log:
+        os.path.join(config["local_log"], APA_SCAN_OUTPREFIX + ".generate_configurationfile.preprocess.log")
+    shell:
+        "( sed " + \
+
+             # Set the rna-seq input directories
+             " 's#input_rnaseq1#" + CURRENT_DIR + "/" + CONDITION1 + "#g; " + \
+             "  s#input_rnaseq2#" + CURRENT_DIR + "/" + CONDITION2 + "#g; " + \
+
+             # Set the PAS-seq input directories
+#             "  s#input_passeq1#" + os.getcwd() + "/test/" + PASSEQ1 + "#g; " + \ # BLANKED OUT AS CAN'T GET THE SCRIPT WORKING WITH PAS-Seq DATA
+#             "  s#input_passeq2#" + os.getcwd() + "/test/" + PASSEQ2 + "#g; " + \ # BLANKED OUT AS CAN'T GET THE SCRIPT WORKING WITH PAS-Seq DATA
+
+             # Set the annotation and genome directories
+             "  s#annotation_name#" + CURRENT_DIR + "/" + ANNOTATION + "#g; " + \
+             "  s#genome_name#" + CURRENT_DIR + "/" + GENOMEFILE + "#g; " + \
+
+             # Set the output directory
+             "  s#out_dir#" + CURRENT_DIR + "/" + os.path.join(config["out_dir"]) + "#g' " + \
+
+             # Input/Output and Log
+             " {input} > {output} ) &> {log} "
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Method-specific rules
+
+
+#### NOTES #####################################################################
+
+# This just runs the APA-Scan.py script, which doesn't take command-line inputs
+# and instead you provide the configuration.ini file which tells the software
+# where to look for the input files (RNA-Seq and PAS-Seq), where the
+# annotation/genome files are found and where to output the result to.
+# because you don't specify input/output from the command line and you don't
+# specify the input files, the configuration.ini file is the ONLY input you
+# need
+
+################################################################################
+
+
+
+rule run_apa_scan:
+    """Run APA-Scan
+    """
+    input:
+        "configuration.ini"
```

Comment:
I think config file should go to main output directory so it's clear it is for a specific run. This would also prevent Snakemake from thinking it has already been generated for subsequent runs (and using the wrong config file for the given run)

i.e. change to `os.path.join(config["out_dir"], "configuration.ini")`